pwiz_tools/Skyline/Model/Results/ChromDataProvider.cs

```diff
@@ -179,6 +179,39 @@ public bool GetChromatogram(int index, out float[] times, out float[] intensitie
 
             return true;
         }
+
+        /// <summary>
+        /// Returns true if the TIC chromatogram present in the .raw file can be relied on
+        /// for the calculation of total MS1 ion current.
+        /// </summary>
+        public bool IsTicChromatogramUsable()
+        {
+            if (!TicChromatogramIndex.HasValue)
+            {
+                return false;
+            }
+
+            float[] times;
+            if (!GetChromatogram(TicChromatogramIndex.Value, out times, out _))
+            {
+                return false;
+            }
+
+            if (times.Length == 0)
+            {
+                return false;
+            }
+
+            // If the number of points in the chromatogram is more than a quarter of the total
+            // spectra, then this chromatogram probably includes MS2 scans, and should not 
+            // be used for calculating TIC Area.
+            if (times.Length >= _dataFile.SpectrumCount / 4)

```

Comment:
I just realized that this would cause us to extract our own TIC for all MSe-type data where every other spectrum is MS1. This may need to change to > count/2.